packages/@react-spectrum/s2/src/SelectBox.tsx

```diff
@@ -0,0 +1,376 @@
+    isSelected: 'gray-900',
+    isFocusVisible: 'blue-900',
+    isDisabled: 'transparent'
+  },
+  transition: 'default',
+  cursor: {
+    default: 'pointer',
+    isDisabled: 'default'
+  },
+  outlineStyle: 'none'
+}, getAllowedOverrides());
+
+const contentContainer = style({
+  display: 'flex',
+  flexDirection: {
+    default: 'column',
+    orientation: {
+      horizontal: 'row'
+    }
+  },
+  justifyContent: 'center',
+  alignItems: 'center',
+  textAlign: {
+    default: 'center',
+    orientation: {
+      horizontal: 'start'
+    }
+  },
+  gap: {
+    default: 8,
+    orientation: {
+      horizontal: 12
+    }
+  },
+  flex: {
+    orientation: {
+      horizontal: '1 0 0'
+    }
+  },
+  width: '100%'
+}, getAllowedOverrides());
+
+const illustrationContainer = style({
+  display: 'flex',
+  alignItems: 'center',
+  justifyContent: 'center',
+  minSize: 48,
+  flexShrink: 0,
+  color: {
+    isDisabled: 'disabled'
+  },
+  opacity: {
+    isDisabled: 0.4
+  }
+});
+
+const textContainer = style({
+  display: 'flex',
+  flexDirection: {
+    orientation: {
+      horizontal: 'column'
+    }
+  },
+  justifyContent: 'center',
+  alignItems: {
+    default: 'center',
+    orientation: {
+      horizontal: 'start'
+    }
+  },
+  gap: {
+    default: 12,
+    orientation: {
+      horizontal: 2
+    }
+  },
+  flex: {
+    orientation: {
+      horizontal: '1 0 0'
+    }
+  },
+  width: '100%',
+  minWidth: 0,
+  color: {
+    default: 'neutral',
+    isDisabled: 'disabled'
+  }
+}, getAllowedOverrides());
+
+const descriptionText = style({
+  display: 'block',
+  alignSelf: 'stretch',
+  width: '100%',
+  minWidth: 0,
+  whiteSpace: 'normal',
+  color: {
+    default: 'neutral',
+    isDisabled: 'disabled'
+  }
+});
+
+const labelText = style({
+  display: 'block',
+  width: '100%',
+  overflow: 'hidden',
+  minWidth: 0,
+  textAlign: {
+    default: 'center',
+    orientation: {
+      horizontal: 'start'
+    }
+  },
+  whiteSpace: 'nowrap',
+  textOverflow: 'ellipsis',
+  alignSelf: {
+    orientation: {
+      horizontal: 'stretch'
+    }
+  },
+  fontWeight: {
+    orientation: {
+      horizontal: 'bold'
+    }
+  },
+  color: {
+    default: 'neutral',
+    isDisabled: 'disabled'
+  }
+});
+
+
+const SelectBoxRenderPropsContext = createContext<{
+  isHovered?: boolean,
+  isFocusVisible?: boolean,
+  isPressed?: boolean
+}>({});
+
+/**
+ * SelectBox components allow users to select options from a list.
+ * Works as content within a ListBoxItem.
+ */
+export const SelectBox = /*#__PURE__*/ forwardRef(function SelectBox(props: SelectBoxProps, ref: FocusableRef<HTMLDivElement>) {
+  [props, ref] = useSpectrumContextProps(props, ref, SelectBoxSpectrumContext);
+  let {
+    children, 
+    value, 
+    isDisabled: individualDisabled = false, 
+    UNSAFE_style
+  } = props;
+  let divRef = useRef<HTMLDivElement | null>(null);
+  let domRef = useFocusableRef(ref, divRef);
+  
+  let contextValue = useContext(SelectBoxContext);
```

Comment:
```suggestion
  let {
    orientation = 'vertical',
    selectedKeys,
    isDisabled: groupDisabled = false,
    isCheckboxSelection = false
  } = useContext(SelectBoxContext);
```

nothing wrong with what you have but you could condense it a bit